apps/myaccount/src/components/shared/system-notification-alert.tsx

```diff
@@ -0,0 +1,207 @@
+                    ),
+                    level: AlertLevels.ERROR,
+                    message: t("myAccount:components.selfSignUp.preference.notifications.genericError.message")
+                }));
+            });
+    };
+
+    /**
+     * Resends the account confirmation Email link.
+     */
+    const resendAccountConfirmationLink = (): void => {
+        if (isSubmitting) {
+            return;
+        }
+        setIsSubmitting(true);
+        resendVerificationLinkOrCode(RecoveryScenario.SELF_SIGN_UP)
+            .then(() => {
+                dispatch(addAlert({
+                    description: t("myAccount:components.systemNotificationAlert.selfSignUp." +
+                        "notifications.resendSuccess.description"),
+                    level: AlertLevels.SUCCESS,
+                    message: t("myAccount:components.systemNotificationAlert.selfSignUp." +
+                        "notifications.resendSuccess.message")
+                }));
+            })
+            .catch((errorMessage: AxiosError) => {
+                dispatch(addAlert({
+                    description: t("myAccount:components.systemNotificationAlert.selfSignUp." +
+                        "notifications.resendError.description", { error: errorMessage }),
+                    level: AlertLevels.ERROR,
+                    message: t("myAccount:components.systemNotificationAlert.selfSignUp." +
+                        "notifications.resendError.message")
+                }));
+            }).finally(() => {
+                setIsSubmitting(false);
+            });
+    };
+
+    /*
+    * The resending capability is available only when account confirmation is configured
+    * to use an Email Link. The rationale for excluding Email OTP and SMS OTP confirmation
+    *  methods is that their respective code validations must occur during the initial
+    * user registration flow itself rather in a later stage via MyAccount.
+    */
+    if (isAccountStatePendingSR && isPreferredChannelEmail && !isSRSendOtpInEmailEnabled) {
+        return (
+            <Alert
+                severity={ AlertLevels.WARNING }
+                data-componentid={ componentId }
+                className="system-notification-alert"
+            >
+                { t("myAccount:components.systemNotificationAlert.selfSignUp." +
+                        "awaitingAccountConfirmation") }
+                <LinkButton
+                    onClick={ resendAccountConfirmationLink }
+                    aria-disabled={ isSubmitting }
+                    disabled={ isSubmitting }
+                    data-testid={ `${ componentId }-resend-link` }
+                >
+                    { t("myAccount:components.systemNotificationAlert.resend") }
+                </LinkButton>
+            </Alert>
+        );
+    }
+
+    return null;
+};
+
+/**
+ * Default properties of {@link SystemNotificationAlert}
+ */
+SystemNotificationAlert.defaultProps = {
```

Comment:
let's avoid using default props for new components.

Ref: https://github.com/wso2/identity-apps/blob/master/docs/write-code/CODE_QUALITY.md#refrain-from-using-defaultprops